src/librustc_mir/monomorphize/collector.rs

```diff
@@ -926,9 +926,10 @@ fn create_mono_items_for_vtable_methods<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
             let poly_trait_ref = principal.with_self_ty(tcx, impl_ty);
             assert!(!poly_trait_ref.has_escaping_bound_vars());
 
-            // Walk all methods of the trait, including those of its supertraits
-            let methods = tcx.vtable_methods(poly_trait_ref);
-            let methods = methods.iter().cloned().filter_map(|method| method)
+            // Walk all methods of the trait, including those of its supertraits.
+            let methods = tcx.vtable_methods(poly_trait_ref).into_iter()
+                .flat_map(|s| s.into_iter());
```

Comment:
```suggestion
                .flat_map(|s| s);
```
identity should just work here because `F: FnMut(Self::Item) -> (U: IntoIterator)`

Reply:
Where's the identity fn in the stdlib, for that matter? :-P

Reply:
It is `std::convert::identity`